core/src/main/java/io/grpc/util/TransmitStatusRuntimeExceptionInterceptor.java

```diff
@@ -160,12 +161,16 @@ public void run() {
 
     @Override
     public void close(final Status status, final Metadata trailers) {
-      serializingExecutor.execute(new Runnable() {
-        @Override
-        public void run() {
-          SerializingServerCall.super.close(status, trailers);
-        }
-      });
+      if (!closeCalled) {
+        closeCalled = true;
+
+        serializingExecutor.execute(new Runnable() {
+          @Override
+          public void run() {
+            SerializingServerCall.super.close(status, trailers);
```

Comment:
`closeCalled` can only be safely read/written from this runnable, which serializes the access in a thread safe way